src/main/java/javax/jmdns/impl/DNSIncoming.java

```diff
@@ -301,12 +307,15 @@ private DNSRecord readAnswer(InetAddress source) {
         String domain = _messageInputStream.readName();
         DNSRecordType type = DNSRecordType.typeForIndex(_messageInputStream.readUnsignedShort());
         if (type == DNSRecordType.TYPE_IGNORE) {
-            logger.warn("Could not find record type. domain: {}\n{}", domain, this.print(true));
+            logger.warn("Could not find record type. domain: {}", domain);
+            if (logger.isDebugEnabled()) {
+                logger.debug("DNSIncoming() unknown type: {}\n{}", type, this.print(true));
+            }
         }
         int recordClassIndex = _messageInputStream.readUnsignedShort();
         DNSRecordClass recordClass = (type == DNSRecordType.TYPE_OPT ? DNSRecordClass.CLASS_UNKNOWN : DNSRecordClass.classForIndex(recordClassIndex));
         if ((recordClass == DNSRecordClass.CLASS_UNKNOWN) && (type != DNSRecordType.TYPE_OPT)) {
-            logger.warn("Could not find record class. domain: {} type: {}\n{}", domain, type, this.print(true));
+            logger.warn("Could not find record class. domain: {} type: {}", domain, type);
```

Comment:
Instead of `type` please log `wireType` (from above comments) as `TYPE_IGNORE` doesn't really help here...

Reply:
Could it happen that for a known resource record the record class is unknown? If yes, then it is probably not always `TYPE_IGNORE`, isn't it?

Reply:
I see only TYPE_OPT being handled explicitly. So some of the newer RRs might be structured in a way that the record class is used for a different purpose AND would result in `TYPE_IGNORE`.